content/en/articles/js/flow/index.md

```diff
@@ -0,0 +1,290 @@
+---
+date: "2016-06-07"
+title: Introduction to Flow, to make your code run on the first try
+tags:
+  - javascript
+  - flow
+  - type
+authors:
+  - bloodyowl
+  - skinnyfoetusboy
+---
+
+```javascript
+document.body.firstChild.getBoundingClientRect()
+```
+
+This hypothetic line of code doesn't always work.
+
+The reason for this is that `element.firstChild` is a `Node` and that not all `Nodes` have a `getBoundingClientRect` method, which is always found on `Elements`, a class that inherits from `Node`.
+
+This kind of error is rather usual, and unfortunately JavaScript doesn't quite get out of its way to warn us that *oi m8 that might not work*. JS actually prefers warning us at *runtime* when it's too late because the bug already happened.
+
+That is because JavaScript is a **weakly/dynamically-typed** language.
+
+Weakly-typed, because if you try to make an operation with two values that have incompatible types, JS will convert those types to ones that it thinks will be the most adequate.
```

Comment:
Consider consolidating these two fragments because they are relating to the same thing.

> That is because JavaScript is a **weakly/dynamically-typed** language; if you try to make an operation with two values that have incompatible types, JS will convert those types to ones that it thinks will be the most adequate.